Whip Reach Fix

Dariensg opened the pull request:
-Made whip only have extra reach on charge attack
-Added parameter to whip to allow for easy changing of the whip, could be hardcoded for more security but I don't see that being an issue
-Removed the UUID for the reach modifier
-Added ability back to break blocks with whip now that the range is back to normal limitations

Dariensg committed:
Whip Reach Fix

-Made whip only have extra reach on charge attack
-Added parameter to whip to allow for easy changing of the whip, could be hardcoded for more security but I don't see that being an issue
-Removed the UUID for the reach modifier
-Added ability back to break blocks with whip now that the range is back to normal limitations

caelwarner commented on src/main/java/com/nekomaster1000/infernalexp/items/IWhipItem.java:
```diff
@@ -1,13 +1,6 @@
 package com.nekomaster1000.infernalexp.items;
```
Do we still actually need this interface? Could we possibly remove it as I don't think it really serves any purpose now

Hellion-Games replied:
That's a good point. I will commit that in a sec

caelwarner reviewed (changes requested):
Just the one question

Dariensg committed:
Removed IWhipItem

-Removed unnecessary interface

caelwarner reviewed (dismissed):
👍

SwanX1 reviewed (changes requested):
I don't think that this approach is at all safe. the client determines the reach distance when sending the packet, which can be unsafe.

Dariensg commented:
We had the same concern about the knockback but decided to leave it as is. I can hard code it on both the client and server side easily, just makes it less convenient to change the reach later. Should I make that change?

SwanX1 commented:
> We had the same concern about the knockback but decided to leave it as is. I can hard code it on both the client and server side easily, just makes it less convenient to change the reach later. Should I make that change?

yes.

Dariensg committed:
Removed Reach from Packet for Security

-Removed the reach value from the packet and hard coded it
-The same can't easily be done with knockback since the knockback is stored on the item, not the player

SwanX1 reviewed (approved):
LGTM,
Would've preferred if you removed the knockback too,
Merge at your own discretion.

Dariensg committed:
Hardcoded Whip Knockback

-Removed Knockback value from being stored in the WhipItem
-Moved the Knockback value to the WhipReachPacket to make it hardcoded for security
-Also clarified instructions for changing the reach distance of the whip
